### DISCUSSION TITLE
Break application when any extension did not load successfully in Production.

### DISCUSSION THREAD

**Discussion by gabrsar** [directus/directus#14011] [Archive]
Hi Guys.

Is there any way to enforce my application to stop if any extension did not load correctly at startup, or if some extension abruptly dies?

We had an extension that didn't was working and that led the system to run without doing what this extension should do, this can be very harmful to production systems to operate partially.

**Comment by haniefss** [comment]
I second this, however I think it's better to have an option for this to choose the way directus behave should any extension fails (keep going or shut down).

**Comment by rijkvanzanten** [comment]
Would you also expect errors thrown in hooks to exit the node process, or should it only prevent server startup? The "extension abruptly dies" part might make this difficult/opinionated

**Reply by diegoleme** [reply]
What if you were treated like other services via healthcheck?

**Reply by rijkvanzanten** [reply]
So the extension would provide a healthcheck function that would be called when the /server/health endpoint is called? That would offer quite a bit of flexibility yeah 🙂

**Reply by gabrsar** [reply]
My main point is that a program is designed to run as a whole, not only some "random" parts of it (in case of unexpected errors for example), and thus is quite important to enforce this in some way.

I believe that only aborting execution is enough. Maybe even giving a callback to handle the exception in loading modules would be enough and not too much opinionated.

I don't yet understand deeply how extensions are integrated inside directus, but them "dying" sounds a bit unusual right? Probably what will happen is the whole process crash.

**Reply by rijkvanzanten** [reply]
> [...] but them "dying" sounds a bit unusual right? Probably what will happen is the whole process crash.

For sure! I was just mostly interested in hearing your thoughts / use case. You mentioned you don't want Directus to startup when an extension fails to load (which makes sense), but that made me curious what you thought should happen in case an extension stops functioning while the server is already running 🙂

**Reply by gabrsar** [reply]
I'm still understanding how directus works under the hood with extensions and so on, so forgive me if I say something silly or wrong.

But as far as I could see, at least in development, the process doesn't run in a monolithic way as it can re-load extensions and so on (I receive a lot of errors while hot-reloading extensions btw, maybe I'm doing something wrong here).

For example, if a filter hook to when a record is created, stops being triggered the application should die because some flow will not run as designed to do. If the same hook has some bug or problem and it always results in an exception being thrown, is ok for the application to stay running as it will trigger some alarm and so on.

A dead process is recognized quickly and it calls the attention of someone that will fix it. As any process partially behaving can stay messing around for days without no one noticing as it just keeps "running", while doing a lot of damage.
